### PR TITLE
Add guidelines for PR reviews.

### PR DESCRIPTION
**Pull request by smoia**
Closes nothing, but follows #181
I tried to write down some guidelines for PR reviewers (and update those for PR authors), as that process could follow guidelines too.
Before even reviewing it, I'd like @rmarkello @eurunuela @vinferrer @RayStick and the rest of the team to throw in more suggestions and discuss these guidelines.

## Proposed Changes

  - Add review guidelines in contributor guide.
  - Minor adjustments related to the adoption of `auto`

### CODE REVIEW

**Comment by RayStick** [comment]
I gave it a read and it looks good to me. At the moment, I have one question relating to reviewing and merging PRs:

If I am the last reviewer to approve a PR, but travis still says "some checks haven't been completed yet" I assume I should wait until I merge? Even if the branch has no conflicts with base branch? If so, we should make this clearer in the contributor file.

**Comment by smoia** [comment]
Yes @RayStick , I added a point about it actually! I pushed, but probably while GitHub was down - I'm trying again.

File: docs/contributorfile.rst [hunk 1]
```diff
@@ -172,11 +182,25 @@ At physiopy, we follow a very similar workflow. The only two differences are:
...
 However, if it’s an incomplete PR, please open a **Draft PR**. That helps us process PRs by knowing which one to have a look first - and how picky to be when doing so.
+
+Reviewing PRs is a time consuming task, that can be stressful for both the reviewer and the author. Avoiding wasting time and the need of little fixes - such as fixing grammar mistakes and typos, styling code, or adopting conventions - is a good start for a successful (and quick) review. Before graduating a Draft PR to a PR ready for review, please check that:
+
+- You did all you wanted to include in your PR. If at a later stage you realise something is missing and it's not minor things, you will need to open a new PR
+- Your code is harmonious with the rest of the code - no repetitions of any sort!
+- Your code respects the `adopted Style <#styling>`_, especially:
+    - your code is lintered adequately and respects the `PEP8 <https://www.python.org/dev/peps/pep-0008/>`_ convention
```
Comment (eurunuela) [inline docs/contributorfile.rst:197]: I'd mention flake8 here.

**Comment by eurunuela** [comment]
I think this is fine. Thanks @smoia !

File: docs/contributorfile.rst (hunk 1 above)
Comment (smoia) [reply docs/contributorfile.rst:197]: I mention it in the style guide already, do you think it is necessary to repeat? Or are you proposing to delete the Style guide?

File: docs/contributorfile.rst [hunk 2]
```diff
@@ -185,13 +209,16 @@ To be merged, PRs have to:
 
 As we’re trying to maintain at least a 90% code coverance, you’re strongly encouraged to write all the necessary tests not to drop below the threshold. If our coverance becomes too low, you might be asked to add more tests and/or your PR might be rejected.
```
Comment (vinferrer) [inline docs/contributorfile.rst:211]: I think we should always ask for a test in cas there is a new feature

Comment (vinferrer) [reply docs/contributorfile.rst:211]: or function

File: docs/contributorfile.rst (hunk 1 above)
Comment (eurunuela) [reply docs/contributorfile.rst:197]: Oh okay, no need to mention it then!

File: docs/contributorfile.rst (hunk 2 above)
Comment (smoia) [reply docs/contributorfile.rst:211]: Maybe I need to rephrase it, but isn't it what we're stating the line before this?

**Comment by RayStick** [comment]
> Yes @RayStick , I added a point about it actually! I pushed, but probably while GitHub was down - I'm trying again.

I see the change now.
Guidelines are clear and easy to read. :)

File: docs/contributorfile.rst (hunk 2 above)
Comment (vinferrer) [reply docs/contributorfile.rst:215]: true

**Comment by vinferrer** [comment]
looks good to me

**Comment by RayStick** [comment]
@smoia - so are we going with what I suggested in the meeting, that the assignee slot means that person is responsible for checking it gets done, and then doing the final merge? 

For the workflow in our lab, I went for these guidelines:

> "Use the ‘Reviewer’ option to assign one (or more) people to review the changes you have made. If the  ‘Assignee’ option is left blank, the last reviewer to approve the PR should do the final merge. If you don’t want this to be the case, use the ‘Assignee’ option to specify which person you want to merge the new branch into master.

If we want something similar, we should add some text explaining this in the docs. I can suggest some.

**Comment by smoia** [comment]
@RayStick yes, I added them in the last commit.
The contributor guidelines document has become huge and probably too complex to be read, if anyone has suggestion on how to improve it, I'd welcome new and fresh ideas!

**Comment by RayStick** [comment]
I can read it properly next week, and try to cut it down a bit, if you like.

**Comment by smoia** [comment]
@RayStick sure, thank you!
I don't know if it's about cutting or about reorganising it, maybe using the other contributor-targetting page, [this one](file:///home/nemo/Scrivania/gitlab/phys2bids/docs/_build/localmedia/contributing.html)

**Comment by smoia** [comment]
@RayStick I'm leaving this as a Draft due to your proposal of rewriting it. Would you have time to do so soon? If not, I'm going to graduate it to PR ready for review and if it's accepted we'll merge it in as it is - we can always work on it at another moment.

**Comment by RayStick** [comment]
I will read it today! Anything specific you want feedback on or improved, or is it just a general read for clarity?

**Comment by smoia** [comment]
> I can read it properly next week, and try to cut it down a bit, if you like.

@RayStick I was referring to this post. It's not necessary short term though! If you think that the documentation is clear and can be merged as is, don't worry about it - just merge it in and we'll think about how to reorganise it at a later time!

**Comment by RayStick** [comment]
@smoia I have submitted a PR to your branch. As well as the changes I suggest there, I also have a few comments:

1. Line179-"_If you see an open issue that you would like to work on, check if it is assigned. If it is, ask the assignee to_" .. this line just ends. :) 
2. It may be worth considering moving the PR Review section directly under the PR section - maybe this is a change for a later time though, not a big deal. 
3. Can the “Main Review” (that is put in the assignee slot) be  the **one** reviewer that is needed? It wasn't clear in the wording that the main reviewer is an extra person or just an extra title, if that makes sense?

I think when I have carried out more of these contributor tasks myself (eventually!) I may have a few more comments about rephrasing of this information to hep newbies get started easily. 

It's good though! Graduate it to a PR!

**Comment by eurunuela** [comment]
@smoia, @RayStick it looks like you have some merge conflicts. How close from review do you think this is?

**Comment by RayStick** [comment]
I'll resolve them now

**Comment by RayStick** [comment]
@eurunuela We are very close to it needing review, but I just want to check a few things with @smoia first.

**Comment by RayStick** [comment]
@smoia I am making a few more smaller changes, that I will submit to your local branch soon.

**Comment by RayStick** [comment]
@smoia let me know if anything is unclear with my most recent changes to these documents. After they've been looked at, then it is ready for review by others.

**Comment by smoia** [comment]
I'm sorry @RayStick , I just missed the PR. @eurunuela do you want to be the main reviewer for this one?

**Comment by RayStick** [comment]
No problem @smoia !

**Comment by eurunuela** [comment]
> I'm sorry @RayStick , I just missed the PR. @eurunuela do you want to be the main reviewer for this one?

Sure, I'll take the lead!

File: docs/contributorfile.rst [hunk 3]
```diff
@@ -192,13 +264,14 @@ Style Guide
...
 
 .. _testing:
 
 Automatic Testing
 -----------------
-physiopy uses Continuous Integration (CI) to make life easier. In particular, we use the `Travis CI <https://travis-ci.org/>`_ platform to run automatic testing!
-**Automatic tests** are cold, robotic, emotionless, and opinionless tests that check that the program is doing what it is expected to. They are written by the developers and run (by Travis CI) every time they send a Pull Request to physiopy repositories. They complement the warm, human, emotional and opinionated **user tests**, as they tell us if a piece of code is failing.
+``physiopy`` uses Continuous Integration (CI) to make life easier. In particular, we use the `Travis CI <https://travis-ci.org/>`_ platform to run automatic testing!
```
Comment (eurunuela) [inline docs/contributorfile.rst:273]: We should probably mention the Azure tests too.

File: docs/contributorfile.rst [hunk 4]
```diff
@@ -57,18 +58,20 @@ When interacting in the common channels, please adhere to our `code of conduct <
 Contributions
 -------------
 
+We welcome and recognize `all contributions <https://allcontributors.org/docs/en/specification>`_ from documentation to testing to code development. You can see a list of current contributors in the README (kept up to date by the `all contributors bot <https://allcontributors.org/docs/en/bot/overview>`_).
```
Comment (eurunuela) [inline docs/contributorfile.rst:61]: Tiny suggestion that I think is easier to read.

```
We welcome and recognize `all contributions <https://allcontributors.org/docs/en/specification>`_ from documentation to testing and code development. You can see a list of current contributors in the README (kept up to date by the `all contributors bot <https://allcontributors.org/docs/en/bot/overview>`_).
```

**Comment by eurunuela** [comment]
Can you please have a look at this @vinferrer and PR?

**Comment by eurunuela** [comment]
Given that this is a documentation change and only one PR is needed, I'll merge if @smoia and @RayStick are happy with it.

**Comment by smoia** [comment]
I'm puzzled by the loss of coverage :thinking:.
In any case, this is a _paused_ PR, since it kind of depends on #181 being merged first.
#181 will be merged as soon as we close #206, #212, #222, #223 - it should happen in two weeks!
If it's accepted, I'll merge it in at the right time. Thank you for keeping an eye on it though!

**Comment by eurunuela** [comment]
I didn't know this PR depended on #181 . Good to know!

**Pull request merged**